app/equipment/weapon.py

```diff
@@ -0,0 +1,10 @@
+from app.contestants.knight import Knight
+
+
+class Weapon:
+    def __init__(self, weapon_power: int) -> None:
+        self.weapon_power = weapon_power
+
+    # apply weapon
+    def apply_weapon(self, knight: Knight) -> None:
+        knight.power += self.weapon_power

```

Comment:
An overall comment regarding the `equipment` module, you create separate classes for each piece of equipment that contain a single method applied to a knight instance. It is not the right approach because when you have a group of methods that act exclusively on the same instance it is the first sign to consider placing them in a class to perform actions over the instance directly. With this approach, you'll avoid a circular import of the `Knight` class and will have a general understanding of all actions that could be applied over a knight instance in terms of equipment.

Reply:
Placed all separate classes, connected with equipment, into the same class. Still don't get how to avoid circular import. Equipment class still have to import from Knight class to perform actions over it. Vice versa, Knight class needs to import from Equipment class in order to apply equipment every time the Knight instance is created.